src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1306,16 +1318,35 @@ private static void ComputeDeclarationsInNode(SemanticModel semanticModel, ISymb
                         executableCodeBlocks = declInNode.ExecutableCodeBlocks;
 
                         // Execute operation actions.
-                        if (shouldExecuteOperationActions && executableCodeBlocks.Any())
+                        if ((shouldExecuteOperationActions || shouldExecuteOperationBlockActions) && executableCodeBlocks.Any())
                         {
-                            var operationsToAnalyze = GetOperationsToAnalyze(executableCodeBlocks, semanticModel, cancellationToken);
-                            foreach (var analyzer in analysisScope.Analyzers)
+                            var operationBlocksToAnalyze = GetOperationBlocksToAnalyze(executableCodeBlocks, semanticModel, cancellationToken);
+                            var operationsToAnalyze = GetOperationsToAnalyze(operationBlocksToAnalyze);
+
+                            if (!operationsToAnalyze.IsEmpty)
                             {
-                                ImmutableDictionary<OperationKind, ImmutableArray<OperationAnalyzerAction>> operationActionsByKind;
-                                if (this.OperationActionsByAnalyzerAndKind.TryGetValue(analyzer, out operationActionsByKind))
+                                if (shouldExecuteOperationActions)
                                 {
-                                    analyzerExecutor.ExecuteOperationActions(operationsToAnalyze, operationActionsByKind,
-                                        analyzer, semanticModel, declarationAnalysisData.TopmostNodeForAnalysis.FullSpan, decl, analysisScope, analysisStateOpt);
+                                    foreach (var analyzer in analysisScope.Analyzers)
+                                    {
+                                        ImmutableDictionary<OperationKind, ImmutableArray<OperationAnalyzerAction>> operationActionsByKind;
+                                        if (this.OperationActionsByAnalyzerAndKind.TryGetValue(analyzer, out operationActionsByKind))
+                                        {
+                                            analyzerExecutor.ExecuteOperationActions(operationsToAnalyze, operationActionsByKind,
+                                                analyzer, semanticModel, declarationAnalysisData.TopmostNodeForAnalysis.FullSpan, decl, analysisScope, analysisStateOpt);
+                                        }
+                                    }
+                                }
+
+                                if (shouldExecuteOperationBlockActions)
+                                {
+                                    foreach (var analyzerActions in GetCodeBlockActions(analysisScope))
```

Comment:
We will end up executing `GetCodeBlockActions` twice if an analyzer has both operation block actions and code block actions. You can probably invoke this upfront if `(shouldExecuteOperationBlockActions || shouldExecuteCodeBlockActions) && executableCodeBlocks.Any()`

Feel free ignore this suggestion as an analyzer having both operation block actions and code block actions is not that likely.